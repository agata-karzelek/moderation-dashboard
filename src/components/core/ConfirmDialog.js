import React from "react";
import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";
import { useStyles } from "../../style/confirmDialog";
import { Typography } from "@material-ui/core";

const ConfirmDialog = (props) => {
  const { title, children, open, setOpen, onConfirm } = props;
  const classes = useStyles();

  return (
    <Dialog
      id="dialog"
      open={open}
      onClose={() => setOpen(false)}
      aria-labelledby="confirm-dialog"
      fullWidth
      maxWidth="xs"
      className={classes.confirmDialog}
    >
      <DialogTitle id="confirm-dialog">{title}</DialogTitle>
      <DialogContent id="content">{children}</DialogContent>
      <DialogActions>
        <Button
          id="confirm"
          onClick={() => {
            setOpen(false);
            onConfirm();
          }}
          className={classes.confirmButton}
        >
          <Typography>{props.actionMessage}</Typography>
        </Button>

        <Button
          id="cancel"
          variant="contained"
          onClick={() => setOpen(false)}
          className={classes.cancelButton}
        >
          Cancel
        </Button>
      </DialogActions>
    </Dialog>
  );
};
export default ConfirmDialog;
