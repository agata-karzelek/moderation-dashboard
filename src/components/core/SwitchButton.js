import Switch from "@material-ui/core/Switch";
import SwitchButtonStyle from "../../style/switchButton";

const SwitchButton = SwitchButtonStyle(({ classes, ...props }) => {
  return (
    <Switch
      focusVisibleClassName={classes.focusVisible}
      disableRipple
      classes={{
        root: classes.root,
        switchBase: classes.switchBase,
        thumb: classes.thumb,
        track: classes.track,
        checked: classes.checked,
      }}
      {...props}
    />
  );
});

export default SwitchButton;
