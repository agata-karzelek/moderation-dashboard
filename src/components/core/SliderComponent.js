import React from "react";
import { useStyles, SlideBar } from "../../style/slideBar";
import { Slider } from "@material-ui/core";

const PrettoSlider = SlideBar(Slider);

export default function SliderComponent({ value, onChange, name }) {
  const classes = useStyles();

  const marks = [
    {
      value: 0,
      label: name === "tisane" ? "None" : 0,
    },
    {
      value: 0.25,
      label: name === "tisane" ? "Extreme" : 0.25,
    },
    {
      value: 0.5,
      label: name === "tisane" ? "High" : 0.5,
    },
    {
      value: 0.75,
      label: name === "tisane" ? "Medium" : 0.75,
    },
    {
      value: 1,
      label: name === "tisane" ? "Low" : 1,
    },
  ];

  function valuetext(values) {
    return `${values}`;
  }

  return (
    <div className={classes.root}>
      <PrettoSlider
        value={value}
        min={0}
        getAriaValueText={valuetext}
        aria-labelledby="discrete-slider"
        step={0.25}
        marks={marks}
        max={1}
        onChange={onChange}
      />
    </div>
  );
}
