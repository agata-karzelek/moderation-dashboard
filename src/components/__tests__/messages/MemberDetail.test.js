import React from "react";
import { shallow, mount } from "enzyme";
import { IconButton } from "@material-ui/core";
import MemberDetail from "../../messages/MemberDetail";

const mockData = {
  block: false,
  mute: false,
  uuid: {
    name: "John",
    profileUrl: "https://www.gravatar.com/avatar/c7b",
    id: "user_00505cca5b04465ca1",
  },
  actionsDisable: false,
  status: "online",
  updated: null,
};
const mockToggleFn = jest.fn();

describe("Test Case For Member Detail", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<MemberDetail member={mockData} toggleMemberDetails={mockToggleFn} />);
  });

  test("Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("MemberDetail Renders", () => {
    expect(wrapper.exists()).toBe(true);
  });

  test("Member Details Card Headings", () => {
    expect(wrapper.find("[testid='nick_name']").text()).toBe("NICK NAME");
    expect(wrapper.find("[testid='created_at']").text()).toBe("CREATED DATE");
    expect(wrapper.find("[testid='email']").text()).toBe("EMAIL ADDRESS");
  });

  test("check functions", () => {
    wrapper = mount(<MemberDetail member={mockData} toggleMemberDetails={mockToggleFn} />);
    wrapper.find("UserAction").getElement().props.updated("", "");

    expect(mockToggleFn).toHaveBeenCalledTimes(0);
    wrapper.find(IconButton).at(0).simulate("click");
    expect(mockToggleFn).toHaveBeenCalledTimes(1);

    expect(wrapper.find("UserAction").getElement().props.action).toBe("");
    expect(wrapper.find("UserAction").getElement().props.open).toBeFalsy();
    wrapper.find(IconButton).at(1).simulate("click");
    expect(wrapper.find("UserAction").getElement().props.open).toBeTruthy();
    expect(wrapper.find("UserAction").getElement().props.action).toBe("mute");

    wrapper.find(IconButton).at(2).simulate("click");
    expect(wrapper.find("UserAction").getElement().props.action).toBe("block");

    wrapper = mount(
      <MemberDetail
        member={{ ...mockData, mute: true, block: true }}
        toggleMemberDetails={mockToggleFn}
      />
    );
    expect(wrapper.find("UserAction").getElement().props.action).toBe("");
    expect(wrapper.find("UserAction").getElement().props.open).toBeFalsy();
    wrapper.find(IconButton).at(1).simulate("click");
    expect(wrapper.find("UserAction").getElement().props.action).toBe("unmute");
    expect(wrapper.find("UserAction").getElement().props.open).toBeTruthy();

    wrapper.find(IconButton).at(2).simulate("click");
    expect(wrapper.find("UserAction").getElement().props.action).toBe("unblock");
  });
});
