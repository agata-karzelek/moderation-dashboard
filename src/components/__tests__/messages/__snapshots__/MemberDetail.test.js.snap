// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Case For Member Detail Snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Grid))
    container={true}
    justify="flex-start"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(IconButton))
            onClick={[Function]}
          >
            <img
              alt="toggle"
              src="/images/Button-Arrow.svg"
            />
          </WithStyles(ForwardRef(IconButton))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <Styled(MuiBox)
            pl={1}
            pt={1}
          >
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-sectionHeading-1"
              variant="body2"
            >
              More Details
            </WithStyles(ForwardRef(Typography))>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <br />
  <WithStyles(ForwardRef(Grid))
    container={true}
    justify="center"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={6}
      sm={6}
    >
      <img
        alt="profile"
        className="makeStyles-profilePicture-2"
        src="https://www.gravatar.com/avatar/c7b"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={12}
      sm={12}
    >
      <WithStyles(ForwardRef(Typography))
        align="center"
        className="makeStyles-name-3"
        variant="h6"
      >
        John
         
        <span
          className="makeStyles-online-9"
        />
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="center"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        lg={3}
        md={4}
        sm={4}
      >
        <WithStyles(WithStyles(ForwardRef(Tooltip)))
          title="Mute"
        >
          <span>
            <WithStyles(ForwardRef(IconButton))
              aria-label="notification"
              disabled={false}
              onClick={[Function]}
              size="small"
            >
              <div
                className="makeStyles-blockIcon-8"
              >
                <img
                  alt="mute"
                  src="/images/unmute.svg"
                  style={
                    Object {
                      "height": "16px",
                      "width": "20px",
                    }
                  }
                />
              </div>
            </WithStyles(ForwardRef(IconButton))>
          </span>
        </WithStyles(WithStyles(ForwardRef(Tooltip)))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        lg={3}
        md={4}
        sm={4}
      >
        <WithStyles(WithStyles(ForwardRef(Tooltip)))
          title="Block"
        >
          <span>
            <WithStyles(ForwardRef(IconButton))
              aria-label="close"
              disabled={false}
              onClick={[Function]}
              size="small"
            >
              <div
                className="makeStyles-blockIcon-8"
              >
                <img
                  alt="block"
                  src="/images/unblock.svg"
                />
              </div>
            </WithStyles(ForwardRef(IconButton))>
          </span>
        </WithStyles(WithStyles(ForwardRef(Tooltip)))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid)) />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-containerRoot-6"
      container={true}
      justify="flex-start"
      spacing={2}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={12}
        sm={12}
      >
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-label-4"
          testid="nick_name"
        >
          NICK NAME
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-labelItem-5"
        >
          John
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={12}
        sm={12}
      >
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-label-4"
          testid="created_at"
        >
          CREATED DATE
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-labelItem-5"
        >
          Invalid date
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={12}
        sm={12}
      >
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-label-4"
          testid="email"
        >
          EMAIL ADDRESS
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-labelItem-5"
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <UserAction
    action=""
    open={false}
    setOpen={[Function]}
    title="Are you sure?"
    updated={[Function]}
    user={Array []}
  >
    You want to 
     this user?
  </UserAction>
</Fragment>
`;
